connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -829,6 +878,34 @@ private boolean startConnector(String connectorName) {
         return started;
     }
 
+    private Callable<Void> getConnectorCallable(final String connectorName, final ConnectorStatus.State targetState) {
+        return new Callable<Void>() {
+            @Override
+            public Void call() throws Exception {
+                switch (targetState) {
+                    case RUNNING:
+                        try {
+                            startConnector(connectorName);
+                        } catch (ConfigException e) {
```

Comment:
Partly true. `Worker` catches `Throwable` but there might be exceptions thrown by the executors right after. Either I'm including the task scheduling of executors in the try block within worker, or I leave this try/catch block here (with `Throwable`)